presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestMapKeyExists.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar.sql;
+
+import com.facebook.presto.operator.scalar.AbstractTestFunctions;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+
+public class TestMapKeyExists
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testBasic()
+    {
+        assertFunction(
+                "MAP_KEY_EXISTS(MAP(ARRAY[1, 2, 3], ARRAY[4, 5, 6]), 2)",
+                BOOLEAN,
+                true);
+        assertFunction(
+                "MAP_KEY_EXISTS(MAP(ARRAY[-1, -2, -3], ARRAY[4, 5, 6]), 2)",
+                BOOLEAN,
+                false);
+        assertFunction(
+                "MAP_KEY_EXISTS(MAP(ARRAY['ab', 'bc', 'cd'], ARRAY['x', 'y', 'z']), 'abc')",
+                BOOLEAN,
+                false);
+        assertFunction(
```

Comment:
Can you add a test case where the key is a NaN?

Reply:
> Can you add a test case where the key is a NaN?

That's probably undefined behavior? Using float/double as map keys is just undefined due to precision issue. So  that won't add any benefit apart from creating a flaky test.

Reply:
While behavior over NaN is broadly undefined, people develop dependencies on undefined behavior, and the test cases actually would be deterministic, so I don't know why the test would be flaky.  BTW, there's a test on reals (without NaN) below.